analyses/cell-type-ewings/exploratory_analysis/05-cluster-exploration.Rmd

```diff
@@ -0,0 +1,626 @@
+      id
+    )
+  })
+```
+
+For both samples we see very clear separation between clusters that express tumor marker genes and each of the normal cell marker genes. 
+
+## Assign cell types to clusters 
+
+Finally, we will assign cell types to all cells in a cluster based on a cluster having high expression of a set of marker genes. 
+
+- Any clusters with high tumor marker gene expression will be labeled "tumor". 
+- Any clusters with high immune marker gene expression also correspond to clusters that are predominantly "macrophage" in the heatmaps, so they will be labeled as "macrophage". 
+- Any clusters with high endothelial marker gene expression also correspond to clusters that are predominantly "endothelial cell" in the heatmaps, so they will be labeled as "macrophage". 
+- There is one cluster in `SCPCL000822` that has high mesenchymal gene expression and is predominantly chondrocytes so we will label this cell as "chondrocytes". 
+- All other clusters with high mesenchymal gene expression appear to have a mix of chondrocytes, fibroblasts, and smooth muscle cells so we will just label these as "mesenchymal-like cells". 
+
+```{r}
+# get umap embeddings for plotting later
+umap_df <- umap_df |>
+  dplyr::select(library_id, barcodes, UMAP1, UMAP2) |>
+  dplyr::distinct()
+
+# assign clusters based on marker gene expression and cell types that are prominent in that cluster
+# use the jaccard matrices to help guide assignments
+cluster_umap_df <- cluster_classification_df |>
+  dplyr::left_join(umap_df, by = c("library_id", "barcodes")) |>
+  dplyr::filter(nn_param == "k_25") |>
+  dplyr::mutate(
+    cluster_celltype = dplyr::case_when(
+      library_id == "SCPCL000822" & cluster %in% c(1, 2, 4, 5, 9) ~ "tumor",
+      library_id == "SCPCL000822" & cluster == 7 ~ "endothelial cell",
+      # cluster 3 has predominantly chondrocytes and has MSC marker gene expression
+      library_id == "SCPCL000822" & cluster == 3 ~ "chondrocyte",
+      # clusters with high "immune" marker gene expression are macrophage
+      library_id == "SCPCL000822" & cluster == 6 ~ "macrophage",
+      # cluster 8 has high MSC marker gene expression and a mix of MSC like cell types
+      library_id == "SCPCL000822" & cluster == 8 ~ "mesenchymal-like cell",
+      library_id == "SCPCL000824" & cluster %in% c(1, 2, 3, 4, 5, 6, 11) ~ "tumor",
+      library_id == "SCPCL000824" & cluster == 8 ~ "endothelial cell",
+      # cluster 10 has high immune markers and is mostly macrohpages
+      library_id == "SCPCL000824" & cluster == 10 ~ "macrophage",
+      # cluster 7 has high MSC marker gene expression and a mix of MSC like cell types
+      library_id == "SCPCL000824" & cluster == 7 ~ "mesenchymal-like cell",
+      # if they didn't have high expression of any markers, we label them as unknown
+      .default = "unknown"
+    ),
+    # make tumor first
+    cluster_celltype = forcats::fct_relevel(cluster_celltype, "tumor", after = 0)
+  )
+```
+
+```{r}
```

Comment:
I'd make this plot bigger, maybe move legend to bottom as needed for spacing. Can't directly suggest changes to the code chunk though since too many backticks...